src/pages/PublicAppointments/PatientSelect.tsx

```diff
@@ -228,6 +164,100 @@ export default function PatientSelect({
           {t("add_new_patient")}
         </Button>
       </div>
+      <div className="flex flex-col justify-center space-y-4">
+        {isLoading ? (
+          <div className="flex justify-center items-center">
+            <Loading />
+          </div>
+        ) : (patients?.length ?? 0) > 0 ? (
+          renderPatientList()
+        ) : (
+          renderNoPatientFound()
+        )}
+      </div>
     </div>
   );
 }
+
+function PatientCard({
+  patient,
+  selectedPatient,
+  setSelectedPatient,
+  getPatienDoBorAge,
+  createAppointment,
+  reason,
+}: {
+  patient: Patient;
+  selectedPatient: string | null;
+  setSelectedPatient: (patientId: string | null) => void;
+  getPatienDoBorAge: (patient: Patient) => string;
+  createAppointment: (body: AppointmentCreateRequest) => void;
+  reason: string;
+}) {
+  const { t } = useTranslation();
+
+  return (
+    <Card
+      key={patient.id}
+      onClick={() => setSelectedPatient(patient.id)}
+      className={cn(
+        "cursor-pointer transition-all duration-200 rounded-xl shadow-md border",
+        selectedPatient === patient.id
+          ? "border-primary shadow-lg"
+          : "hover:border-gray-300",
+      )}
+    >
+      <CardHeader>
+        <CardTitle className="capitalize text-lg font-semibold">
+          {patient.name}
+        </CardTitle>
+      </CardHeader>
+      <CardContent className="space-y-2">
+        <div className="flex justify-between">
+          <span className="text-sm text-muted-foreground font-medium">
+            {t("date_of_birth_age")}:
+          </span>
+          <span className="text-sm font-semibold">
+            {getPatienDoBorAge(patient)}
+          </span>
+        </div>
+        <div className="flex justify-between">
+          <span className="text-sm text-muted-foreground font-medium">
+            {t("sex")}:
+          </span>
+          <span className="text-sm font-semibold">
+            {t(`GENDER__${patient.gender}`)}
+          </span>
+        </div>
+
+        {selectedPatient === patient.id && (
+          <div className="mt-4 flex flex-row gap-3">
+            <Button
+              variant="destructive"
+              className="w-1/2 md:w-auto"
+              onClick={(e) => {
+                e.stopPropagation();
+                setSelectedPatient(null);
+              }}
+            >
+              {t("cancel")}
+            </Button>
+            <Button
+              variant="primary"
+              className="w-1/2 md:w-auto"
+              onClick={(e) => {
+                e.stopPropagation();
+                createAppointment({
+                  patient: patient.id ?? "",
+                  reason_for_visit: reason ?? "",
+                });
+              }}
+            >
+              {t("confirm")}
+            </Button>
+          </div>
+        )}
```

Comment:
instead of keeping the buttons inside the patient card, can't we keep it outside in the page itself? 🤔